pynicotine/shares.py

```diff
@@ -578,24 +643,26 @@ def __init__(self):
         self.requested_share_times = {}
         self.pending_network_msgs = []
         self.rescanning = False
-        self.should_compress_shares = False
-        self.compressed_shares = {}
-
-        for share_type in ("public", "buddy"):
-            self.compressed_shares[share_type] = slskmessages.SharedFileListResponse()
+        self.compressed_shares = {
+            "public": slskmessages.SharedFileListResponse(),
+            "buddy": slskmessages.SharedFileListResponse(),
+            "trusted": slskmessages.SharedFileListResponse(),
+            "banned": slskmessages.SharedFileListResponse()
```

Comment:
Unclear of "banned" purpose in here?

Reply:
If a banned user requests your shares, they get an empty list (this used to be in the _shared_file_list_request method before, but I needed it for local shares as well).